pkg/controllers/generic/from_host_syncer.go

```diff
@@ -0,0 +1,318 @@
+	}
+	return found
+}
+
+func (s *genericFromHostSyncer) ExcludePhysical(_ client.Object) bool {
+	return false
+}
+
+func (s *genericFromHostSyncer) enqueuePhysical(ctx *synccontext.SyncContext, obj client.Object, q workqueue.TypedRateLimitingInterface[ctrl.Request]) {
+	if obj == nil {
+		return
+	}
+	if nn, ok := s.shouldSync(ctx, obj); ok {
+		q.Add(reconcile.Request{NamespacedName: nn})
+	}
+}
+
+func (s *genericFromHostSyncer) shouldSync(_ *synccontext.SyncContext, obj client.Object) (types.NamespacedName, bool) {
+	hostName, hostNs := obj.GetName(), obj.GetNamespace()
+	if _, ok := obj.GetLabels()[translate.MarkerLabel]; ok {
+		// do not sync objects that were synced from virtual to host already
+		return types.NamespacedName{}, false
+	}
+	return s.MatchesHostObject(hostName, hostNs)
+}
+
+func (s *genericFromHostSyncer) checkExperimentalDeployConfig(ctx *synccontext.RegisterContext) map[string]bool {
+	// (Pawel): this is needed, because when user sets:
+	// sync.toHost.(configMaps/secrets).enabled = true
+	// sync.toHost.(configMaps/secrets).all = true
+	// sync.fromHost.(configMaps/secrets).enabled = true
+	// sync.fromHost.(configMaps/secrets).mappings: (mapping from vCluster host namespace to NAMESPACE_A)
+	// experimental.deploy.vCluster.manifests: (manifest for (config map / secret) in NAMESPACE_A)
+	// then, fromHost syncer will enqueue (config map/ secret) coming from experimental.deploy.vCluster.manifests
+	// syncer will try to call SyncToHost, as physical object does not exist
+	// fromHostSyncer.SyncToHost will delete virtual object, which is not wanted and expected by user.
+	// Therefore, we need to exclude objects from experimental.deploy.vCluster.(manifests/manifestsTemplate)
+	// in the genericFromHostSyncer.
+	deploy := ctx.Config.Experimental.Deploy
+	if deploy.VCluster.Manifests == "" && deploy.VCluster.ManifestsTemplate == "" {
+		return nil
+	}
+	virtualConfigMapsToSkip := make(map[string]bool)
+	if strings.Contains(deploy.VCluster.Manifests, "---") {
+		for _, manifest := range strings.Split(deploy.VCluster.Manifests, "---") {
+			configMapKey, found := s.processManifest(manifest)
+			if found {
+				virtualConfigMapsToSkip[configMapKey] = true
+			}
+		}
+	} else {
+		configMapKey, found := s.processManifest(deploy.VCluster.Manifests)
+		if found {
+			virtualConfigMapsToSkip[configMapKey] = true
+		}
+	}
+
+	if strings.Contains(deploy.VCluster.ManifestsTemplate, "---") {
+		for _, manifest := range strings.Split(deploy.VCluster.ManifestsTemplate, "---") {
+			configMapKey, found := s.processTemplate(manifest, &ctx.Config.Config, ctx.Config.Name, ctx.Config.WorkloadTargetNamespace)
+			if found {
+				virtualConfigMapsToSkip[configMapKey] = true
+			}
+		}
+	} else {
+		configMapKey, found := s.processTemplate(deploy.VCluster.ManifestsTemplate, &ctx.Config.Config, ctx.Config.Name, ctx.Config.WorkloadTargetNamespace)
+		if found {
+			virtualConfigMapsToSkip[configMapKey] = true
+		}
+	}
```

Comment:
You shouldn't need the `strings.Contains` check.  If it doesn't contain "---" the split will return a single element slice, so it could be more like 

```go
	for _, manifest := range strings.Split(deploy.VCluster.Manifests, "---") {
		configMapKey, found := s.processManifest(manifest)
		if found {
			virtualConfigMapsToSkip[configMapKey] = true
		}
	}

	for _, manifest := range strings.Split(deploy.VCluster.ManifestsTemplate, "---") {
		configMapKey, found := s.processTemplate(manifest, &ctx.Config.Config, ctx.Config.Name, ctx.Config.WorkloadTargetNamespace)
		if found {
			virtualConfigMapsToSkip[configMapKey] = true
		}
	}
```

Reply:
good point, changed